File: old/actions/set_postgres_password.py
"""
Set PostgreSQL password action
"""
import logging
from .base import Action
logger = logging.getLogger(__name__)

class SetPostgresPasswordAction(Action):
    """Action to set PostgreSQL password"""
    description = "postgresql password setup"

    def execute(self) -> bool:
        """Set PostgreSQL password"""
        if not self.ssh_service:
            logger.error("SSH service not initialized")
            return False
        # Get password from params
        password = "postgres"
        if hasattr(self, "container_cfg") and self.container_cfg:
            params = self.container_cfg.params or {}
            password = params.get("password", "postgres")
        # Use local socket connection (requires local entry in pg_hba.conf)
        command = f"sudo -n -u postgres psql -c \"ALTER USER postgres WITH PASSWORD '{password}';\" 2>&1"
        output, exit_code = self.ssh_service.execute(command, sudo=False)
        if exit_code is not None and exit_code != 0:
            logger.error("set postgres password failed with exit code %s", exit_code)
            if output:
                logger.error("set postgres password output: %s", output.splitlines()[-1])
            return False
        return True

File: src/actions/set_postgres_password.go
package actions

import (
	"fmt"
	"strings"
	"enva/libs"
	"enva/services"
)

// SetPostgresPasswordAction sets PostgreSQL password
type SetPostgresPasswordAction struct {
	*BaseAction
}

func NewSetPostgresPasswordAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &SetPostgresPasswordAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *SetPostgresPasswordAction) Description() string {
	return "postgresql password setup"
}

func (a *SetPostgresPasswordAction) Execute() bool {
	if a.SSHService == nil {
		libs.GetLogger("set_postgres_password").Printf("SSH service not initialized")
		return false
	}
	password := "postgres"
	if a.ContainerCfg != nil && a.ContainerCfg.Params != nil {
		if p, ok := a.ContainerCfg.Params["password"].(string); ok {
			password = p
		}
	}
	command := fmt.Sprintf("sudo -n -u postgres psql -c \"ALTER USER postgres WITH PASSWORD '%s';\"", password)
	output, exitCode := a.SSHService.Execute(command, nil)
	if exitCode != nil && *exitCode != 0 {
		libs.GetLogger("set_postgres_password").Printf("set postgres password failed with exit code %d", *exitCode)
		if output != "" {
			lines := strings.Split(output, "\n")
			if len(lines) > 0 {
				libs.GetLogger("set_postgres_password").Printf("set postgres password output: %s", lines[len(lines)-1])
			}
		}
		return false
	}
	return true
}

